src/util/animate.js

```diff
@@ -7,17 +7,17 @@
    * @typedef {Object} AnimationOptions
    * @property {Function} [options.onChange] Callback; invoked on every value change
    * @property {Function} [options.onComplete] Callback; invoked when value change is completed
-   * @property {Number} [options.startValue=0] Starting value
-   * @property {Number} [options.endValue=100] Ending value
-   * @property {Number} [options.byValue=100] Value to modify the property by
+   * @property {number | number[]} [options.startValue=0] Starting value
+   * @property {number | number[]} [options.endValue=100] Ending value
+   * @property {number | number[]} [options.byValue=100] Value to modify the property by
```

Comment:
Should we write somewhere here or on top of the block that:
- this is done to animate arrays, in particular matrices ( unless you have other examples )
- that you have to be consistent, either all arrays or all single numbers? ( if you have to. i m not sure )

Reply:
Added an example note that should clarify that they should be consistent and also the values they can take. Currently if the start value is an array then all the other values `startValue`, `endValue`, and `byValue` should match.